packages/functions/src/donor/ConfirmAppointmentHandler.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  getAppointmentsByIds,
+  ConfirmArrivedFromDbAppointment,
+  setAppointment
+} from "../dal/AppointmentDataAccessLayer";
+import { FunctionsApi } from "@zm-blood-components/common";
+import { getDonor } from "../dal/DonorDataAccessLayer";
+import { NotificationToCoordinator } from "../notifications/NotificationSender";
+import { notifyOnAppointmentWithType } from "../notifications/SendAppointmentNotifier";
+
+export default async function(
+  request: FunctionsApi.ConfirmAppointmentRequest,
+  callerId: string
+) {
+  const donorId = callerId;
+
+  if (!request.appointmentId) {
+    throw new Error("No appointment to confirm");
+  }
+
+  const appointmentToConfirm = await getAppointmentsByIds([
+    request.appointmentId
+  ]);
+  if (appointmentToConfirm.length !== 1) {
+    throw new Error("Appointment not found");
+  }
+
+  const appointment = appointmentToConfirm[0];
+  if (appointment.donorId !== donorId) {
+    throw new Error("Appointment to be confirmed is not booked by donor");
+  }
+
+  const donor = await getDonor(donorId);
+  notifyOnAppointmentWithType(appointment, donor!, NotificationToCoordinator.APPOINTMENT_CONFIRMED).catch((e) =>
```

Comment:
Let's handle the notifications in a different PR.

We need to create a template for it, and add the template id to the repo

Reply:
commenting it out...